app/views/users/new.haml

```diff
@@ -7,26 +7,26 @@
       %tr#signup-username
         %th.sub.vtop= f.label :username
         %td.even
-          = f.text_field :username, placeholder: "Username"
+          = f.text_field :username, placeholder: "Username", autofocus: true, autocomplete: "username", minlength: User::MIN_USERNAME_LEN, maxlength: User::MAX_USERNAME_LEN, required: true
           .user-alert.hidden
             = image_tag 'icons/exclamation.png', alt: '!', title: '', class: 'vmid'
             %span.msg
       %tr#signup-email
         %th.sub.vtop= f.label :email
         %td.odd
-          = f.text_field :email, placeholder: "Email address"
+          = f.text_field :email, placeholder: "Email address", autocomplete: "email", type: "email", required: true
           .user-alert.hidden
             = image_tag 'icons/exclamation.png', alt: '!', title: '', class: 'vmid'
             %span.msg
       %tr#signup-password
         %th.sub.vtop= f.label :password
         %td.even
-          = f.password_field :password, placeholder: "Password"
+          = f.password_field :password, placeholder: "Password", autocomplete: "new-password", minlength: User::MIN_PASSWORD_LEN, required: true
           .user-alert.hidden
             = image_tag 'icons/exclamation.png', alt: '!', title: '', class: 'vmid'
             %span.msg
       %tr#signup-password-confirmation
-        %th.sub.vtop= f.label :password_confirmation, 'Confirm'
+        %th.sub.vtop= f.label :password_confirmation, 'Confirm', autocomplete: "new-password", minlength: User::MIN_PASSWORD_LEN, required: true
         %td.odd
           = f.password_field :password_confirmation, placeholder: "Confirm Password"
```

Comment:
Very belatedly (aka i just noticed this while rebasing onto this) I think something went wrong here

Reply:
https://github.com/glowfic-constellation/glowfic/pull/2366 :D?